pilot/pkg/xds/eds.go

```diff
@@ -193,6 +197,13 @@ func (s *DiscoveryServer) edsCacheUpdate(shard model.ShardKey, hostname string,
 				newIstioEndpoints = append(newIstioEndpoints, nie)
 			}
 		}
+		// Next, check for endpoints that were in old but no longer exist. If there are any, there is a
+		// removal so we need to push an update.
+		for _, oie := range oldIstioEndpoints {
+			if _, f := nmap[oie.Address]; f {
```

Comment:
Sorry I am bit confused. Should this be `!f`?

Reply:
That is a good point... now I am wondering how the test passed...

Reply:
Ah lol, it passes because one endpoint is `f` and one if `!f`. Let me make the test more robust so it fails with the current code